File: Sources/Model/MA2FolderTaxonomy.swift
import Foundation

/// grandMA2 USB layout, rooted at `<volume>/gma2/`.
///
/// Verified via MA Lighting documentation (2026-08-19): the `gma2` folder
/// contains `shows`, `bitmaps`, `colors`, `effects`, `gobos`, and
/// `importexport` at its top level. Unlike grandMA3's `gma3_library`, the
/// exact per-object-type subfolder names *inside* `importexport` (macros,
/// presets, sequences, etc.) could not be confirmed from documentation
/// alone — rather than guess and risk wrong shortcuts, this taxonomy only
/// covers the confirmed top-level folders plus a generic "Import/Export"
/// entry the user can drill into manually. If you verify the exact
/// `importexport` subfolder names against a real MA2 USB stick, add them
/// here the same way `MA3FolderTaxonomy` breaks down `datapools/*`.
enum MA2FolderTaxonomy {
    static let categories: [ConsoleCategory] = [
        ConsoleCategory(id: "ma2_shows", displayName: "Shows", symbolName: "theatermasks", rootKind: .library,
                    relativePath: RelativePath(components: ["shows"])),
        ConsoleCategory(id: "ma2_importexport", displayName: "Import/Export", symbolName: "square.and.arrow.up.on.square", rootKind: .library,
                    relativePath: RelativePath(components: ["importexport"])),
        ConsoleCategory(id: "ma2_effects", displayName: "Effects", symbolName: "sparkles", rootKind: .library,
                    relativePath: RelativePath(components: ["effects"])),
        ConsoleCategory(id: "ma2_gobos", displayName: "Gobos", symbolName: "circle.grid.cross", rootKind: .library,
                    relativePath: RelativePath(components: ["gobos"])),
        ConsoleCategory(id: "ma2_bitmaps", displayName: "Bitmaps", symbolName: "square.grid.3x3.square", rootKind: .library,
                    relativePath: RelativePath(components: ["bitmaps"])),
        ConsoleCategory(id: "ma2_colors", displayName: "Colors", symbolName: "paintpalette", rootKind: .library,
                    relativePath: RelativePath(components: ["colors"])),
    ]
}

File: GrandMAResourceManager/Models/MA2FolderTaxonomy.cs
namespace GrandMAResourceManager.Models;

/// <summary>
/// grandMA2 layout. USB root: <c>&lt;volume&gt;\gma2\</c>. Windows onPC root:
/// <c>C:\ProgramData\MA Lighting Technologies\grandMA\gma2_V_x.y.z\</c> (a
/// different, version-scoped instance of this same relative layout — see
/// <c>Services/LocalInstallDetector.cs</c> and <c>Sources/DiskFileSource.cs</c>).
///
/// Verified via MA Lighting documentation (2026-08-19): the folder contains
/// <c>shows</c>, <c>bitmaps</c>, <c>colors</c>, <c>effects</c>, <c>gobos</c>,
/// and <c>importexport</c> at its top level. The exact per-object-type
/// subfolder names *inside* <c>importexport</c> (macros, presets,
/// sequences, ...) could not be confirmed from documentation alone — rather
/// than guess and risk wrong shortcuts, this taxonomy only covers the
/// confirmed top-level folders plus a generic "Import/Export" entry to
/// drill into manually. Update this file once verified against a real MA2
/// installation/USB stick, the same way <see cref="MA3FolderTaxonomy"/>
/// broke down <c>datapools/*</c> after checking real data.
/// </summary>
public static class MA2FolderTaxonomy
{
    public static readonly IReadOnlyList<ConsoleCategory> Categories = new[]
    {
        new ConsoleCategory("ma2_shows", "Shows", Glyphs.Document, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "shows" })),
        new ConsoleCategory("ma2_importexport", "Import/Export", Glyphs.Folder, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "importexport" })),
        new ConsoleCategory("ma2_effects", "Effects", Glyphs.Folder, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "effects" })),
        new ConsoleCategory("ma2_gobos", "Gobos", Glyphs.Folder, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "gobos" })),
        new ConsoleCategory("ma2_bitmaps", "Bitmaps", Glyphs.Folder, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "bitmaps" })),
        new ConsoleCategory("ma2_colors", "Colors", Glyphs.Folder, ConsoleCategoryRootKind.Library, new RelativePath(new[] { "colors" })),
    };
}
